[docs] Add Release Notes document (#1274)
Azure/iotedge-lorawan-starterkit: docs/release-notes.md

---
title: Release Notes
---

## v2.0.0

### New Features

> 

Reviewer: - Support for [LoRa Basics™ Station](https://github.com/lorabasics/basicstation/tree/v2.0.5) LNS Protocol (v2.0.5)
- Support for [LoRa Basics™ Station](https://github.com/lorabasics/basicstation/tree/v2.0.5) CUPS Protocol (v2.0.5) for credential management and firmware upgrade
- LoRaWAN Network Server and Basics™ Station can be decoupled, allowing the possibility to run multiple concentrators on a single LoRaWAN Network Server instance
- Support for secure communication between LoRaWAN Network Server and Basics™ Station
- Support for running LoRaWAN Network Server on Azure IoT Edge for Linux on Windows
- Regional support for AS923 and CN470 (Regional Parameters v1.0.3rA and RP002-1.0.3 revisions)
- Observability: Integration with Azure Monitor
- Observability: exposing a Prometheus endpoint for metrics scraping
- Observability: @bastbu please add more here 

I would also add a "Other" or "Code quality" section with:
- Introduction of LoRaWAN Primitives and refactoring of existing code to make wide use of those
- Upgrade to .NET 6 (LTS version)
- Upgrade container base images to Debian 11
- Improvements in E2E CI, continuously testing multiple scenarios and authentication modes
- @p-schuler is there something we want to write here or in the breaking changes regarding the device caching?

Reviewer 2: > LoRaWAN Network Server and Basics™ Station can be decoupled, allowing the possibility to run multiple concentrators on a single LoRaWAN Network Server instance

Correction: allowing the possibility to **connect** multiple concentrators to a single LNS.

Reviewer 2: There are not breaking changes to the device caching. It's been changed and improved, but not breaking compatibility.

Reviewer 2: Observability: default monitoring for edgeAgent and edgeHub modules.

Reviewer 2: Observability: distributed tracing

> - Support for multiple Azure IoTEdge Gateways with cache and deduplication

Reviewer 3: ```suggestion
- Support for multiple Azure IoT Edge Gateways with cache and deduplication
```

> - Support for multiple Azure IoT Edge Gateways with cache and deduplication

Reviewer 2: we always supported multi gateway.

- Support for multiple concentrators with cache and deduplication
- Support for running LNS on a separate Edge or Cloud device

> - [EFLOW](https://docs.microsoft.com/en-us/azure/iot-edge/iot-edge-for-linux-on-windows?view=iotedge-2018-06) Compatibility

Reviewer 3: ```suggestion
- [Support for LNS on Windows](https://docs.microsoft.com/en-us/azure/iot-edge/iot-edge-for-linux-on-windows?view=iotedge-2018-06) Compatibility
```

- Support for [LoRa Basics™ Station](https://github.com/lorabasics/basicstation)

> - Regional support for CN470 and AS923

Reviewer 3: Should we be explicit about RP1 and RP2 for china?

- Support for CUPS credential management
- Support for CUPS concentrator firmware update
- First class support for Class C devices
- Observability: integration with Azure Monitor and metrics that are exposed on a Prometheus endpoint

> ### Breaking Changes

Reviewer: In this section I would at least add:

- Infrastructure changes: New storage containers for CUPS credentials management and firmware upgrade (more in related [ADRs](link to ADRs)
- Infrastructure changes: @bastbu were there some changes related to deployment of Azure Monitor resources?

Reviewer 2: We need log analytics for the integration with the metrics collector.

---
title: Release Notes
---

## v2.0.0

### New Features

- Support for [LoRa Basics™ Station](https://github.com/lorabasics/basicstation)
LNS Protocol (v2.0.5)
- Support for [LoRa Basics™ Station](https://github.com/lorabasics/basicstation)
CUPS Protocol (v2.0.5) for credential management and firmware upgrade
- LoRaWAN Network Server and Basics™ Station can be decoupled, allowing the
possibility to connect multiple concentrators on a single LoRaWAN Network Server
instance
- Support for secure communication between LoRaWAN Network Server and Basics™
Station
- Support for running LoRaWAN Network Server on [Azure IoT Edge for Linux on
Windows](https://docs.microsoft.com/en-us/azure/iot-edge/iot-edge-for-linux-on-windows?view=iotedge-2018-06)
- Regional support for AS923 and CN470 (Regional Parameters v1.0.3rA and
RP002-1.0.3 revisions)
- Support for running LNS on a separate Edge or Cloud device

> - First class support for Class C devices

Reviewer: ```suggestion
```
I would remove this sentence. We already had Class C support before

- Observability:
  - Integration with Azure Monitor
  - Exposing a Prometheus endpoint for metrics scraping

>   - default monitoring for edgeAgent and edgeHub modules

Reviewer: ```suggestion
  - Collection of metrics for edgeAgent and edgeHub modules with [Metrics collector module](https://docs.microsoft.com/en-us/azure/iot-edge/how-to-collect-and-transport-metrics?view=iotedge-2020-11&tabs=iothub)
```

>   - distributed tracing

Reviewer: We should be more detailed on what this means

Reviewer 4: I think we should remove it, we don't really have support for distributed tracing.

### Code Quality Improvements

- Introduction of LoRaWAN Primitives and refactoring of existing code to make
wide use of those
- Upgrade to .NET 6 (LTS version)
- Upgrade container base images to Debian 11
- Improvements in E2E CI, continuously testing multiple scenarios and
authentication modes

### Breaking Changes

- Deprecated support for Packetforwarder ([upgrade instructions](user-guide/pkt-fwd-to-station.md))
- Upgrade to RaspberryPi OS based on Debian 11.0 Bullseye ([upgrade instructions](user-guide/upgrade.md#upgrading-to-raspberry-pi-os-bullseye))
- Infrastructure changes:
  - New storage containers for CUPS credentials management and firmware upgrade
  (more in related [ADRs](adr/008_CUPS_firmware_upgrade.md))
  - A Log Analytics workspace is required for integrating with metrics collector
  (more in related [ADRs](adr/005_observability.md))

### Bugfixes

- [#249](https://github.com/Azure/iotedge-lorawan-starterkit/issues/249)
At least one resource deployment operation failed
- [#310](https://github.com/Azure/iotedge-lorawan-starterkit/issues/310)
Incorrect data type for the tmms property

## v2.0.0-beta

> - certificate validation

Reviewer: This is a breaking change:

Compared to v2.0.0-alpha, in v2.0.0-beta we start validating not only the certificate thumbprint (as per device twin) but also that the chain of trust is correct. Therefore, any user is required to follow the steps [here](https://azure.github.io/iotedge-lorawan-starterkit/dev/user-guide/station-authentication-modes/?h=chain#changing-client-certificate-mode-in-lorawan-network-server-module-and-trusting-certificate-chain)

---
title: Release Notes
---

## v2.0.0

### New Features

- Support for [LoRa Basics™ Station](https://github.com/lorabasics/basicstation)
LNS Protocol (v2.0.5)
- Support for [LoRa Basics™ Station](https://github.com/lorabasics/basicstation)
CUPS Protocol (v2.0.5) for credential management and firmware upgrade
- LoRaWAN Network Server and Basics™ Station can be decoupled, allowing the
possibility to connect multiple concentrators on a single LoRaWAN Network Server
instance
- Support for secure communication between LoRaWAN Network Server and Basics™
Station
- Support for running LoRaWAN Network Server on [Azure IoT Edge for Linux on
Windows](https://docs.microsoft.com/en-us/azure/iot-edge/iot-edge-for-linux-on-windows?view=iotedge-2018-06)
- Regional support for AS923 and CN470 (Regional Parameters v1.0.3rA and
RP002-1.0.3 revisions)
- Support for running LNS on a separate Edge or Cloud device

> - Observability:

Reviewer 4: ```suggestion
- [Observability](./user-guide/observability.md):
```

  - Integration with Azure Monitor
  - Exposing a Prometheus endpoint for metrics scraping
  - Collection of metrics for edgeAgent and edgeHub modules with [Metrics collector module](https://docs.microsoft.com/en-us/azure/iot-edge/how-to-collect-and-transport-metrics?view=iotedge-2020-11&tabs=iothub)
  - distributed tracing

### Code Quality Improvements

- Introduction of LoRaWAN Primitives and refactoring of existing code to make
wide use of those
- Upgrade to .NET 6 (LTS version)
- Upgrade container base images to Debian 11
- Improvements in E2E CI, continuously testing multiple scenarios and
authentication modes

### Breaking Changes

- Deprecated support for Packetforwarder ([upgrade instructions](user-guide/pkt-fwd-to-station.md))
- Upgrade to RaspberryPi OS based on Debian 11.0 Bullseye ([upgrade instructions](user-guide/upgrade.md#upgrading-to-raspberry-pi-os-bullseye))

> - Infrastructure changes:

Reviewer: Let's add that the Azure Function extension version needs to be updated to ~4 to support v2.0.0

Reviewer 2: maybe link the upgrade instructions in general rather than repeating that information: https://azure.github.io/iotedge-lorawan-starterkit/dev/user-guide/upgrade/
